src/main/DAL/MetaQuestion/MetaQuestionRepository.js

```diff
@@ -44,12 +80,78 @@ class MetaQuestionRepository {
      * @returns The updated question.
      */
     async addAnswersToQuestion(qId, answers) {
-        await answers.forEach(answer => {
+        answers.forEach(answer => {
             answer.metaQuestionId = qId;
         });
         await this.#Answer.bulkCreate(answers);
 
-        return await this.#MetaQuestion.findByPk(qId, {
+        return await this.getMetaQuestion(qId);
+    }
+
+    /**
+     * Adds given keywords to given meta-question.
+     * @param qId ID of an existing meta-question.
+     * @param keywords Array of keywords: string[]
+     * @returns The updated question.
+     */
+    async addKeywordsToQuestion(qId, keywords) {
+        const metaQuestion = await this.getMetaQuestion(qId);
+        const dbKeywords = await this.#getDbKeywords(keywords)
+        await metaQuestion.addKeywords(dbKeywords);
+        await metaQuestion.reload();
+        return metaQuestion;
+    }
```

Comment:
what if keywords is not already saved in the db ? does getDbKeywords save them?

metaQuestion.addKeywords add them ?

Same goes with addKeywordsToAppendix....

Reply:
yes exactly that. getDbKeywords creates rows for the given keywords (unless they already exist). 
once you have an array of DB elements the method `addKeywords` (which is generated through the relation definition automatically) creates the necessary connections in the DB